src/Symfony/Component/Serializer/Context/Encoder/XmlEncoderContextBuilder.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Context\Encoder;
+
+use Symfony\Component\Serializer\Context\ContextBuilder;
+use Symfony\Component\Serializer\Encoder\XmlEncoder;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * A helper providing autocompletion for available XmlEncoder options.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+class XmlEncoderContextBuilder extends ContextBuilder
+{
+    /**
+     * Configures whether the decoded result should be considered as a collection
+     * or as a single element.
+     */
+    public function withAsCollection(?bool $asCollection): static
+    {
+        return $this->with(XmlEncoder::AS_COLLECTION, $asCollection);
+    }
+
+    /**
+     * Configures node types to ignore while decoding.
+     *
+     * @see https://www.php.net/manual/en/dom.constants.php
+     *
+     * @param list<int>|null $decoderIgnoredNodeTypes
+     *
+     * @throws InvalidArgumentException
+     */
+    public function withDecoderIgnoredNodeTypes(?array $decoderIgnoredNodeTypes): static
+    {
+        foreach ($decoderIgnoredNodeTypes ?? [] as $decoderIgnoredNodeType) {
+            if (!\is_int($decoderIgnoredNodeType)) {
```

Comment:
Not sure if it's necessary. IMO providing the PHPDoc is enough, if the user cares about that it should use Psalm or PHPStan. Same for all these checks.